tkp/main.py

```diff
@@ -19,70 +20,82 @@
                             group_per_timestep
                             )
 from tkp.db.configstore import store_config, fetch_config
-from tkp.steps.persistence import create_dataset, store_images
+from tkp.steps.persistence import create_dataset, store_images_in_db
 import tkp.steps.forced_fitting as steps_ff
 from tkp.steps.varmetric import execute_store_varmetric
 
 
 logger = logging.getLogger(__name__)
 
 
-def run(job_name, supplied_mon_coords=[]):
+def setup(job_name, supplied_mon_coords=None):
+    """
+    Initialises the pipeline run.
+    """
+    if not supplied_mon_coords:
+        supplied_mon_coords = []
+
     pipe_config = initialize_pipeline_config(
         os.path.join(os.getcwd(), "pipeline.cfg"),
         job_name)
 
     # get parallelise props. Defaults to multiproc with autodetect num cores
-    parallelise = pipe_config.get('parallelise', {})
-    distributor = os.environ.get('TKP_PARALLELISE', parallelise.get('method',
-                                                                    'multiproc'))
-    runner = Runner(distributor=distributor,
-                    cores=parallelise.get('cores', 0))
-
+    distributor = os.environ.get('TKP_PARALLELISE', pipe_config.parallelise.method)
+    runner = Runner(distributor=distributor, cores=pipe_config.parallelise.cores)
 
-    #Setup logfile before we do anything else
+    # Setup logfile before we do anything else
     log_dir = pipe_config.logging.log_dir
-    setup_logging(log_dir,
-                  debug=pipe_config.logging.debug,
-                  use_colorlog=pipe_config.logging.colorlog
-                  )
+    setup_logging(log_dir, debug=pipe_config.logging.debug,
+                  use_colorlog=pipe_config.logging.colorlog)
 
     job_dir = pipe_config.DEFAULT.job_directory
     if not os.access(job_dir, os.X_OK):
         msg = "can't access job folder %s" % job_dir
         logger.error(msg)
-        raise IOError(msg)
+        sys.exit(1)
```

Comment:
I'm quite against calling `sys.exit` if you're not in handler code in the very top level of the command line script - although it seems unlikely right now, some future version of this 'setup' function might be called as part of a larger Python script - maybe a loop which runs TraP on several different datasets in succession, for example - and so in that case you prefer to have the option of handling an exception rather than being forced to a hard exit. 

Also, IIRC the IOError trace will typically include details of which file it couldn't find, which is useful to have in the logs.